### PR TITLE
Remote refactor

### PR DESCRIPTION
**Pull request by kn-ms**
## Bug / Requirement Description
Some refactoring of the remote pool/worker related code

## Solution description
Describe your code changes in detail for reviewers.
- The WorkerBase now serves as a base for Workers as Worker was doing to much
- UnboundRemoteResourceConfig that represent a remote resource configuration without a port assigned for possible
  remote resources that got their assignement async after construction time.
- environment now do not try to stop aborted resources

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/common/entity/base.py [hunk 1]
```diff
@@ -1498,7 +1505,7 @@ def run(self):
             raise self._runnable.result
         return self._runnable.result
 
-    def _handle_abort(self, signum):
+    def _handle_abort(self, signum, frame):
```
Comment (Pyifan) [inline testplan/common/entity/base.py:1508]: what is frame for?

**Review by Pyifan — CHANGES_REQUESTED** [review]

**Comment by Pyifan** [comment]
hi Krisztian, I'm not sure about the now even more complicated class hierarchy, is there any room for simplification? Otherwise this looks good and I'm going to look at the internal PR next.

File: testplan/common/entity/base.py (hunk 1 above)
Comment (kn-ms) [reply testplan/common/entity/base.py:1508]: That is expected from a signal handler: https://docs.python.org/3/library/signal.html#signal.signal

**Comment by kn-ms** [comment]
> hi Krisztian, I'm not sure about the now even more complicated class hierarchy, is there any room for simplification? Otherwise this looks good and I'm going to look at the internal PR next.

With the old remote worker config you were not able to capture configuration for remote worker without a set hostname:port. This make it difficult to use the remote worker in a composition where the remote worker can be recreated on different hosts from a previously captured config as hosts joining and leaving the plant/run

**Review by Pyifan — APPROVED** [review]

**Pull request merged**